examples/preprocessing/plot_fnirs_artifact_removal.py

```diff
@@ -0,0 +1,74 @@
+"""
+==========================================
+Visualise NIRS artifact correction methods
+==========================================
+
+Here we artificially introduce several fNIRS artifacts and observe
+how artifact correction techniques attempt to correct the data.
+
+"""
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+import os
+import mne
+
+###############################################################################
+# Import data
+# -----------
+#
+# Here we will work with the :ref:`fNIRS motor data <fnirs-motor-dataset>`.
+# We resample the data to make indexing exact times for convenient.
+# We then convert the data to optical density to perform corrections on.
+
+fnirs_data_folder = mne.datasets.fnirs_motor.data_path()
+fnirs_raw_dir = os.path.join(fnirs_data_folder, 'Participant-1')
+raw_intensity = mne.io.read_raw_nirx(fnirs_raw_dir, verbose=True).load_data()
+raw_intensity.resample(3, npad="auto")
+raw_od = mne.preprocessing.nirs.optical_density(raw_intensity)
+
+
+###############################################################################
+# Add artificial artifacts to data
+# --------------------------------
+#
+# We add a spike type artifact at 100 seconds and a baseline shift at 200
+# seconds to the data. These are both common types of artifacts in NIRS data.
+
+corrupted_od = raw_od.copy()
+corrupted_od._data[:, 298:302] = corrupted_od._data[:, 298:302] - 0.06
+corrupted_od._data[:, 600:900] = corrupted_od._data[:, 600:900] + 0.03
```

Comment:
Modifying private attributes in examples is frowned upon. How about something like:
```
corrupted_data = raw.get_data()
corrupted_data[:, 298:302] = ...
...
corrupted_od = mne.io.RawArray(corrupted_data, raw_od.info, first_samp=raw_od.first_samp)
```
?